getretake/retake
# refactor: use pgsync for real time sync

**Ticket(s) Closed**

- Closes #

**What**

This PR refactors the real time stack to use pgsync instead of kafka to synchronize data from postgres to opensearch.

**Why**

While the kafka stack worked well, it was also much more complex in terms of deploy and had a higher resource requirements. As such, the team decided to make this change to reduce complexity by using pgsync, which we also aim to improve.


**How**
To test this PR:
1. Start the stack with `docker compose up`
2. Send a request to the `realtime/link` endpoint to start processing. you will need to pass a source object which includes connection details, and a schema as defined in the [pgsync documentation](https://pgsync.com/schema/):

```bash
curl --location 'http://localhost:8000/index/realtime/link' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer retake-test-key' \
--data '{
    "source": {
        "index_name": "customers",
        "source_host": "postgres",
        "source_port": "5432",
        "source_user": "postgres",
        "source_password": "postgres",
        "source_relation": "movies",
        "source_primary_key": "postgres",
        "source_columns": [
            "id",
            "name",
            "description"
        ]
    },
    "pgsync_schema": {
        "database": "postgres",
        "index": "movies",
        "nodes": {
            "table": "movies",
            "schema": "public",
            "columns": [
                "id",
                "name",
                "description"
            ]
        }
    }
}'
```
3. Once you get a successful response, you can see pgsync running in the background with `docker compose logs pgsync`.
4. Finally, check your index in opensearch to see if the rows were indexed successfully.

**Tests**
None yet
